### PR TITLE
Client side routing to improve the navigation times for "View Deposit History"

### LINKED ISSUE #6527: Deposits card "View full deposits history" button should use client-side navigation for improved UX
### Describe the bug

Clicking the "View full deposits history" button on the Payments → Overview card causes a full browser page load when redirecting. We should use client-side navigation to improve this user-experience.

### To Reproduce

1. Ensure you have at least 1 product purchased via this store.
2. Go to **Payments → Overview** and find the **Deposits** card.
3. Click on the "View full deposits history" button.
4. Wait as the browser redirects to the deposits list screen, via a full-page reload.
5. On the **Payments → Deposits** screen, click an individual deposit. Notice this uses client-side navigation for a smoother experience.

### Actual behavior

Clicking the "View full deposits history" button on the Payments → Overview card causes a full browser page load when redirecting.

### Screenshots

image

### Expected behavior

Clicking the "View full deposits history" button on the Payments → Overview card should use client-side navigation and not result in a full browser page load.

### Desktop (please complete the following information):

* OS: macOS
* Browser: Arc
* Version: Version 0.107.0

### Additional context

This can be achieved using the `@woocommerce/navigation` library:

```ts
import { getHistory } from '@woocommerce/navigation';
const history = getHistory();
history.push(
	getAdminUrl( {
		page: 'wc-admin',
		path: '/payments/deposits',
	} )
);
```

### PR DESCRIPTION
**Pull request by naman03malhotra**
Fixes #6527 

#### Changes proposed in this Pull Request

- Used the`getHistory` hook to leverage the client-side navigation. 
- Create a common function that tracks the events and navigate to the `/payments/deposits` page.

#### Testing instructions

- Verify that on clicking the "View deposits history" button, it navigates via the client side and doesn't reload the whole page while navigating.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `8375` or branch name `fix/6527-improved-navigation-time` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/6527-improved-navigation-time) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 84fa8e14f55345b90715a982814390e81c7212a3
- Build time: 2024-03-15 06:57:07 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/components/deposits-overview/index.tsx [hunk 1]
```diff
@@ -73,6 +74,18 @@ const DepositsOverview: React.FC = () => {
 		account?.deposits_schedule?.interval
 	);
 
+	const navigateToDepositsHistory = () => {
+		recordEvent( 'wcpay_overview_deposits_view_history_click' );
+
+		const history = getHistory();
+		history.push(
```
Comment (Jinksi) [inline client/components/deposits-overview/index.tsx:88]: ⛏️🤏 tiny nitpick: is it better or worse to make this more concise?
```suggestion
		getHistory().push(
```

Comment (naman03malhotra) [reply client/components/deposits-overview/index.tsx:88]: I had the same thought, and then I went ahead with what is the most common way to use it, but I think for this case, we can lean either way.

**Review by Jinksi — APPROVED** [review]
Nice one @naman03malhotra ✨ 

I've tested this and it works well to make the navigation from Payments Overview → Deposits list screen fast and smooth.

A downside of this is that it doesn't obey CMD+click for opening in a new browser tab. This may be a tradeoff worth making, so happy to approve.

There is no straightforward solution for this with the `Button` component (`Link` handles this nicely, however). I recall bumping into this previously and we ended up wrapping `Button` in a `Link` – not ideal, but it allowed CMD+click and fast navigation.

https://github.com/Automattic/woocommerce-payments/blob/8ddca7fdd57bc8e1a8879f125cfe105af15db115/client/payment-details/dispute-details/dispute-resolution-footer.tsx#L241-L259

**Comment by naman03malhotra** [comment]
>Link handles this nicely, however

I also considered this use case, which I have seen handled in open-source UI Kits, where developers create custom components to style Link components as buttons leveraging the react router navigation, which provides the best of both worlds.

**Comment by naman03malhotra** [comment]
Some tests were failing due snapshots, updated here: https://github.com/Automattic/woocommerce-payments/pull/8375/commits/47d83bf6793425c343f416465a099ebf479de855

**Comment by haszari** [comment]
> There is no straightforward solution for this with the Button component (Link handles this nicely, however). I recall bumping into this previously and we ended up wrapping Button in a Link – not ideal, but it allowed CMD+click and fast navigation.

Curious why we didn't follow that pattern here? I thought we'd landed on that as a good and straightforward solution. 😄

**Pull request merged**